### PR TITLE
fix: show loading banner when blockaid toggled

### PR DESCRIPTION
**Pull request by blackdevelopa**
## **Description**

When ppom is initialised on mobile, we notice that some devices experiences performance issues within the first couple seconds. Having this loading banner helps improve this experience for these devices.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1769

## **Manual testing steps**

1. Go to Settings
2. Go to the Experimental
3. Toggle blockaid on

### **After**

https://github.com/MetaMask/metamask-mobile/assets/29962968/28492809-42f6-4a55-b6e1-021e7f66bfe8

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by brianacnguyen — DISMISSED** [review]

File: app/actions/experimental/index.ts [hunk 1]
```diff
@@ -3,9 +3,17 @@ import type { Action } from 'redux';
...
 
 export interface SetSecurityAlertsEnabled
   extends Action<ActionType.SET_SECURITY_ALERTS_ENABLED> {
   securityAlertsEnabled: boolean;
 }
+
+export function SetPpomInitializedCompleted(value: boolean) {
+  return {
+    type: ActionType.SET_PPOM_INITIALIZATION_COMPLETED,
+    ppomInitializationCompleted: value,
```
Comment (jpuri) [inline app/actions/experimental/index.ts:17]: `ppomInitializationCompleted` looks redundant here.

File: app/actions/experimental/index.ts [hunk 2]
```diff
@@ -3,9 +3,17 @@ import type { Action } from 'redux';
...
 export enum ActionType {
   SET_SECURITY_ALERTS_ENABLED = 'SET_SECURITY_ALERTS_ENABLED',
+  SET_PPOM_INITIALIZATION_COMPLETED = 'SET_PPOM_INITIALIZATION_COMPLETED',
 }
 
 export interface SetSecurityAlertsEnabled
   extends Action<ActionType.SET_SECURITY_ALERTS_ENABLED> {
   securityAlertsEnabled: boolean;
 }
+
```
Comment (jpuri) [inline app/actions/experimental/index.ts:13]: may be `SetPPOMInitializedCompleted`

File: app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    InteractionManager.runAfterInteractions(() => {
+      AnalyticsV2.trackEvent(
+        MetaMetricsEvents.SETTINGS_EXPERIMENTAL_SECURITY_ALERTS_ENABLED,
+        {
+          security_alerts_enabled: !initialSecurityAlertsEnabled,
+        },
+      );
+    });
+  };
+
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx:70]: Props like `title`, `description` do no require to be passed unless that can change depending on parent state.

File: package.json [hunk 4]
```diff
@@ -189,7 +189,7 @@
     "@metamask/network-controller": "^8.0.0",
     "@metamask/permission-controller": "^4.0.1",
     "@metamask/phishing-controller": "^5.0.0",
-    "@metamask/ppom-validator": "^0.11.0",
+    "@metamask/ppom-validator": "0.13.0",
     "@metamask/preferences-controller": "^4.0.0",
```
Comment (jpuri) [inline package.json:193]: PR needs to be rebased.

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 5]
```diff
@@ -21,110 +12,53 @@ import { getNavigationOptionsTitle } from '../../../UI/Navbar';
...
-    PreferencesController?.setSecurityAlertsEnabled(!securityAlertsEnabled);
-    setSecurityAlertsEnabled(!securityAlertsEnabled);
-    InteractionManager.runAfterInteractions(() => {
-      AnalyticsV2.trackEvent(
-        MetaMetricsEvents.SETTINGS_EXPERIMENTAL_SECURITY_ALERTS_ENABLED,
-        {
-          security_alerts_enabled: !securityAlertsEnabled,
-        },
-      );
-    });
```
Comment (jpuri) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:125]: Metrics code above looks like is removed.

File: app/components/Views/Settings/ExperimentalSettings/index.tsx [hunk 6]
```diff
@@ -21,110 +12,53 @@ import { getNavigationOptionsTitle } from '../../../UI/Navbar';
...
-  const [securityAlertsEnabled, setSecurityAlertsEnabled] = useState(
-    () => PreferencesController.state.securityAlertsEnabled,
+  const dispatch = useDispatch();
+
+  const initialSecurityAlertsEnabled = useSelector(
+    selectIsSecurityAlertsEnabled,
+  );
+  const [securityAlertsEnabled, setSecurityAlertsEnabled] = useState<any>(
+    initialSecurityAlertsEnabled,
   );
```
Comment (jpuri) [inline app/components/Views/Settings/ExperimentalSettings/index.tsx:34]: We may not need a duplicate local state variable here.

File: app/components/Views/Settings/ExperimentalSettings/index.tsx (hunk 5 above)
Comment (blackdevelopa) [reply app/components/Views/Settings/ExperimentalSettings/index.tsx:125]: It was moved to the another file [here](https://github.com/MetaMask/metamask-mobile/pull/8013/files#diff-09f0d5b4b498f3af62029a8ee7f7b302b46cdf2a1f2fa2b00f69f4c7a965c6f1R69)

File: app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
+import React, { useRef, useEffect, useState } from 'react';
```
Comment (segun) [inline app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx:1]: Why is this component inside experimental settings? Maybe it should probably be in components/UI ?

File: app/components/Views/Settings/ExperimentalSettings/ExperimentalSettings.styles.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
+import { StyleSheet } from 'react-native';
```
Comment (segun) [inline app/components/Views/Settings/ExperimentalSettings/ExperimentalSettings.styles.ts:1]: Some of these styles are applied to the blockaidIndicator component, not the ExperimentalSettings component, so we should move the styles for blockaidIndicator to it's own file.

File: app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx (hunk 7 above)
Comment (segun) [reply app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx:1]: I also think the name should start with a capital letter.

**Review by segun — COMMENTED** [review]
Left a few comments

File: app/components/Views/Settings/ExperimentalSettings/ExperimentalSettings.styles.ts (hunk 8 above)
Comment (blackdevelopa) [reply app/components/Views/Settings/ExperimentalSettings/ExperimentalSettings.styles.ts:1]: Thank you. Updated [here](https://github.com/MetaMask/metamask-mobile/pull/8013/commits/290806fad8b00693f04021b5f7f022112a30b91a)

File: app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx (hunk 7 above)
Comment (blackdevelopa) [reply app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx]: Thank you. Updated [here](https://github.com/MetaMask/metamask-mobile/pull/8013/commits/290806fad8b00693f04021b5f7f022112a30b91a)

File: app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+    (state: any) => state.experimentalSettings.ppomInitializationCompleted,
+  );
+
+  useEffect(() => {
+    if (ppomInitialisationCompleted) {
+      setIntialisedBlockaid(false);
+      setFetchingPPOMDataInProgress(false);
+      setSheetInteractable(true);
+    }
+  }, [ppomInitialisationCompleted]);
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply app/components/Views/Settings/ExperimentalSettings/blockaidIndicator.tsx]: Thank you. Updated [here](https://github.com/MetaMask/metamask-mobile/pull/8013/commits/290806fad8b00693f04021b5f7f022112a30b91a)

**Review by segun — DISMISSED** [review]

File: app/core/Engine.ts [hunk 10]
```diff
@@ -702,6 +702,12 @@ class Engine {
           securityAlertsEnabled:
             initialState.PreferencesController?.securityAlertsEnabled ?? false,
           state: initialState.PPOMController,
+          ppomInitialisationCallback: (): any => {
+            store.dispatch({
+              type: 'SET_PPOM_INITIALIZATION_COMPLETED',
+              ppomInitializationCompleted: true,
+            });
+          },
```
Comment (cryptotavares) [inline app/core/Engine.ts:710]: This feels like an anti-pattern.. Ideally we would like to keep the engine as decoupled as possible from the application state. IMO we should rely on the controller state updates, that would trigger an update in the application redux state (those state changes are propagated in the EngineService).

Comment (jpuri) [reply app/core/Engine.ts:710]: Yep that is good point, I am planning updates to PPOMController post release and we should add this point to it.

**Review by jpuri — DISMISSED** [review]

**Review by jpuri — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Review by brianacnguyen — APPROVED** [review]
Approved changes to DS

**Pull request merged**